### PR TITLE
added support for v3.1-RC

### PR DESCRIPTION
**Pull request by Alessandro100**
Added support for GBFS v3.1-RC

New custom rules added
- [system_pricing_plans] `reservation_price_per_min` or `reservation_price_flat_rate` allowed in plans
- [vehicle_types] if `reservation_price_per_min` or `reservation_price_flat_rate` is present in the associated vehicle type, then `default_reserve_time` becomes required

Key changed files
gbfs-validator/versions/partials/v3.1-RC/vehicle_types/default_reserve_time_require.js
gbfs-validator/gbfs.js

Key test files
gbfs-validator/__test__/fixtures/conditional_default_reserve_time.js
gbfs-validator/__test__/gbfs.test.js

References
https://gbfs.org/specification/
https://github.com/MobilityData/gbfs/blob/v3.1-RC/gbfs.md

### CODE REVIEW

File: gbfs-validator/versions/schemas/v3.1-RC/system_pricing_plans.json [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+              "name",
+              "currency",
+              "price",
+              "is_taxable",
+              "description"
+            ],
+            "dependencies": {
+              "reservation_price_flat_rate": { "not": { "required": ["reservation_price_per_min"] } },
+              "reservation_price_per_min": { "not": { "required": ["reservation_price_flat_rate"] } }
+            }
```
Comment (Alessandro100) [inline gbfs-validator/versions/schemas/v3.1-RC/system_pricing_plans.json:196]: @richfab I added this to the schema which says: either `reservation_price_flat_rate` or  `reservation_price_per_min` is present but not both of them together. Let me know if an addition like this to the schema is possible or should this be done in the form of a validation patch

Comment (richfab) [reply gbfs-validator/versions/schemas/v3.1-RC/system_pricing_plans.json:196]: Nice improvement! Great use of the JSON schema features. I'll add it to https://github.com/MobilityData/gbfs-json-schema. Thanks 🙏

Comment (richfab) [reply gbfs-validator/versions/schemas/v3.1-RC/system_pricing_plans.json:196]: PR is up for review: https://github.com/MobilityData/gbfs-json-schema/pull/125

**Review by richfab — APPROVED** [review]
LGTM 💯 
Thanks @Alessandro100 🙏

File: gbfs-validator/versions/schemas/v3.1-RC/system_pricing_plans.json (hunk 1 above)
Comment (richfab) [reply gbfs-validator/versions/schemas/v3.1-RC/system_pricing_plans.json:196]: Yes, currently the schemas in https://github.com/MobilityData/gbfs-validator/ are duplicated from https://github.com/MobilityData/gbfs-json-schema/.

Open to suggestions to make sure they are always in sync.

Comment (testower) [reply gbfs-validator/versions/schemas/v3.1-RC/system_pricing_plans.json:196]: Two possible ways among others:

* download them from the other repo (how it's done in entur validator)
* use git submodule

Comment (richfab) [reply gbfs-validator/versions/schemas/v3.1-RC/system_pricing_plans.json:196]: Create an issue for this: https://github.com/MobilityData/gbfs-validator/issues/186

**Pull request merged**